src/mesh/structured_mesh.jl

```diff
@@ -0,0 +1,15 @@
+struct StructuredMesh{RealT<:Real, NDIMS}
```

Comment:
```suggestion
struct StructuredMesh{NDIMS, RealT<:Real}
```
If you change the order of the parameters, it will be easier to create something like 
```julia
WeNeedToFindAGoodNameForThisGuy{NDIMS} = Union{TreeMesh{NDIMS}, StructuredMesh{NDIMS}}
```
which could be used to simplify code.

Reply:
We didn't know you can dispatch by the first argument. That's pretty cool. Thanks.